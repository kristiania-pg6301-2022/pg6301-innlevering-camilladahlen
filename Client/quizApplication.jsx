import { useLoader } from "./useLoader";
import { fetchJSON, postJSON } from "./http";
import React, { useState } from "react";

function QuizComponent({ reload }) {
  const [question, setQuestion] = useState();
  const [answer, setAnswer] = useState();

  async function handleLoadQuestion() {
    setQuestion(await fetchJSON("/api/question"));
  }

  function handleReload() {
    setQuestion(undefined);
    setAnswer(undefined);
    reload();
  }

  if (!question) {
    return (
      <div>
        <button onClick={handleLoadQuestion}>Load a new question</button>
      </div>
    );
  }

  return (
    <QuestionDisplay
      question={question}
      answer={answer}
      setAnswer={setAnswer}
      onReload={handleReload}
    />
  );
}

///This function actually renders the question with all the answers etc.
function QuestionDisplay({ question, answer, setAnswer, onReload }) {
  async function handleAnswer(answer) {
    const result = await postJSON("/api/question", { id: question.id, answer });
    setAnswer(await result.text());
  }

  if (!answer) {
    return (
      <div>
        <h2>{question.question}</h2>
        {Object.keys(question.answers)
          .filter((a) => question.answers[a])
          .map((a) => (
            <div key={a}>
              <button onClick={() => handleAnswer(a)}>
                {question.answers[a]}
              </button>
            </div>
          ))}
      </div>
    );
  }
  return (
    <div>
      <p>You answered {answer === "true" ? "correctly" : "incorrect"}</p>
      <button onClick={onReload}>New question</button>
    </div>
  );
}

export function FrontPage() {
  //const { reload } = useLoader(async () => fetchJSON("/"));
  return (
    <div>
      <h1>Quiz-app!</h1>
      {/*<QuizComponent reload={reload} />*/}
    </div>
  );
}
