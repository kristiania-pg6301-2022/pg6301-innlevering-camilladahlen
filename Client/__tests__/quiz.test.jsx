import React from "react";
import ReactDOM from "react-dom";
import { FrontPage } from "../quizApplication";

describe("quiz application", () => {
  it("shows a question", () => {
    const element = document.createElement("div");
    ReactDOM.render(<FrontPage />, element);
    expect(element.innerHTML).toMatchSnapshot();
  });
});
