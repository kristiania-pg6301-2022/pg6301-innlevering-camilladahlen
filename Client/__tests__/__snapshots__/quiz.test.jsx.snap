// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quiz application shows a question 1`] = `"<div><h1>Quiz-app!</h1></div>"`;
